### PR TITLE
Windows support for T2V scripts

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
@Nojahhh Would you mind giving this a review? You've already fixed the dataset side of things with lambda pickling error, and this PR replaces the inline collate_fn with a class instance as per your PR #32 in the remaining scripts

### CODE REVIEW

File: training/cogvideox_text_to_video_lora.py [hunk 1]
```diff
@@ -199,6 +198,26 @@ def log_validation(
     return videos
 
 
+class CollateFunction:
```
Comment (sayakpaul) [inline training/cogvideox_text_to_video_lora.py:201]: Mind explaining the change?

**Review by sayakpaul — APPROVED** [review]
Looks minimal .

**Comment by Nojahhh** [comment]
@a-r-r-o-w I think it looks good. Will try it out and report back if I encounter any errors.

File: training/cogvideox_text_to_video_lora.py (hunk 1 above)
Comment (Nojahhh) [reply training/cogvideox_text_to_video_lora.py:201]: Inline function raised an error with pickle in windows, moving collate function to a class instances solves the issue.

**Review by Nojahhh — APPROVED** [review]
All three training methods have been tested and work as intended.

**Comment by a-r-r-o-w** [comment]
Thanks for all your help @Nojahhh!

**Pull request merged**